hedera-node/src/main/java/com/hedera/services/ledger/BalanceChange.java

```diff
@@ -233,4 +244,11 @@ public void setExemptFromCustomFees(boolean exemptFromCustomFees) {
 	public boolean isExemptFromCustomFees() {
 		return exemptFromCustomFees;
 	}
+
+	public boolean hasOnlyAlias() {
+		AutoAccountsManager autoAccounts = AutoAccountsManager.getInstance();
```

Comment:
Using a static method for object lookup is an anti-pattern; imo we can:

1. Let Dagger2 create the `AutoAccountsManager` as a singleton in the normal way.
2. Inject that singleton into the `TransferLogic`.
3. Pass it into this `BalanceChange.hasOnlyAlias()` method.